### ISSUE TITLE
Put `webgpu` behind a feature flag

### ISSUE THREAD

**Issue by atbrakhi** [servo/servo#34341]
This should help us reduce the binary size. After the initial investigation, it looks like it should help us reduce the size by at least 5MB.

**Comment by sagudev** [comment]
Sounds good, most of webgpu is standalone anyway. There are only connection with canvas and obtaining device/adapter. Oh and AsyncWGPU stuff.

**Comment by atbrakhi** [comment]
@sagudev  I noticed that the `GPUCanvasContext` rendering context type for WebGPU rendering is tightly coupled in the [`HTMLCanvasElement.Webidl`](https://github.com/servo/servo/blob/19a7e95a6a4cc4e6aa187f2fe2302c4707401e34/components/script/dom/webidls/HTMLCanvasElement.webidl#L9) file, which makes adding the webgpu flag in the HTML canvas element tricky.

I thought maybe I could try to separate `GPUCanvasContext` into a separate file, but then I noticed that the HTML [spec](https://html.spec.whatwg.org/multipage/canvas.html#dom-canvas-getcontext) requires `getContext()` to handle all rendering contexts, including webgpu. This might explain why the current implementation keeps them coupled. However, this would also mean that if I proceed with separating `GPUCanvasContext`, I would be deviating from the spec, and ideally, we should avoid that.

I'm wondering if there is any other way to ensure we follow the spec while also keeping webgpu as separated as possible?
